### ISSUE TITLE
Custom routes not accessible by anonymous users

### ISSUE THREAD

**Issue by jeremydrichardson** [marmelab/react-admin#7489]
**What you were expecting:**
Custom routes would be accessible by users who are _not_ logged in.

**What happened instead:**
When trying to access a custom route directly, it automatically redirects you back to the login page unless you are logged in.

**Steps to reproduce:**
Enter the url of the custom page directly in the url bar when not logged in. eg. `/#/mypage`

**Related code:**
https://codesandbox.io/s/react-adminv4-customroute-noaccess-h5sdg6

```
<CustomRoutes noLayout>
  <Route path="/mypage" element={<MyPage />} />
</CustomRoutes>
```

**Other information:**

**Environment**

* React-admin version: 4.0.0-rc.1
* Last version that did not exhibit the issue (if applicable): 4.0.0-rc.0
* React version: 17
* Browser: Chrome
* Stack trace (in case of a JS error):

**Comment by slax57** [comment]
Reproduced, thanks for submitting this!

**Comment by fzaninotto** [comment]
Fixed by #7491

**Comment by jeremydrichardson** [comment]
> Reproduced, thanks for submitting this!

Glad to be able to help out a bit!
